### PR TITLE
Усова - Лабораторная работа #1

### PR DESCRIPTION
**Pull request by UsovaMA**
@AnastasiaaSenina @JuliaKamelina please review

### CODE REVIEW

File: code/my_set/model/my_set.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        else:
+            position = self.my_set.index(element)
+            del self.my_set[position]
+
+    def union(self, obj):
+        if isinstance(obj, list):
+            for element in obj:
+                self.add(element)
+        elif isinstance(obj, MySet):
+            self.union(obj.my_set)
```
Comment (JuliaKamelina) [inline code/my_set/model/my_set.py:58]: This part can be simplified
```suggestion
        if isinstance(obj, (list, MySet)):
            for element in obj:
                self.add(element)
```

File: code/my_set/model/my_set.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
+class MySet:
```
Comment (AnastasiaaSenina) [inline code/my_set/model/my_set.py:1]: Please rename `MySet` -> `Set`

File: code/my_set/model/my_set.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    def __init__(self, elements=None):
+        self.my_set = list()
+        if isinstance(elements, list):
+            self.my_set = elements
+        elif elements is None:
+            pass
+        elif isinstance(elements, int):
+            self.my_set.append(elements)
+        else:
+            raise TypeError('Input error: argument type must be list')
```
Comment (AnastasiaaSenina) [inline code/my_set/model/my_set.py:12]: Please change this logic. First check that `elements is not None`.

File: code/my_set/model/my_set.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        if isinstance(obj, MySet):
+            result = MySet()
+            for element in self:
+                if element not in obj:
+                    result.add(element)
+            return result
+        else:
+            raise TypeError('Input error: argument type for operations on sets must be only set')
+
+    def issubset(self, obj):
```
Comment (AnastasiaaSenina) [inline code/my_set/model/my_set.py:83]: Please choose one style for name of method : `is_empty()`, `is_subset()` or `isempty()`, `issubset()` .  `is_empty()`, `is_subset()` better, IMHO.

**Review by JuliaKamelina — APPROVED** [review]
👍

**Comment by vzinovie** [comment]
👍

**Comment by UsovaMA** [comment]
@AnastasiaaSenina done, check please

**Review by AnastasiaaSenina — APPROVED** [review]
👍

**Comment by UsovaMA** [comment]
@kirill-korniakov

**Pull request merged**